addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/internal/utility/Holiday.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.tankerkoenig.internal.utility;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+
+/**
+ * Utility static class to calculate if a given date is a german holiday.
+ * Presently only those holidays that are valid in whole germany (Bundesweit) are used!
+ * Code copied from Openhab1-Addons Samples-Rules
+ *
+ * @author Jürgen Baginski
+ *
+ */
+public final class Holiday {
+
+    private Holiday() {
+        // avoid instantiation.
+    }
+
+    public static boolean isHoliday(DateTime now) {
```

Comment:
I agree that this logic should not be hidden here within the binding code - such a feature would actually fit to the discussion at https://github.com/eclipse/smarthome/pull/3587, wdyt?
@JueBag Could we live for a start with NOT supporting holidays in this binding and add that feature as soon as we have platform support available for that?